### PR TITLE
Added 118. Pascal's Triangle solution in java

### PR DESCRIPTION
**Pull request by akshay399**
# Pull Request Template

## Description

Please include a summary of the problem and about the approach to solve the solution. Please also include relevant motivation and context.
List any dependencies that are required for this change.

## Have you made changes in [README](https://github.com/codedecks-in/LeetCode-Solutions/blob/master/README.md) file ?
- [x] Added problem & solution under correct topic.
- [x] Specified Space & Time complexity.
- [x] Specified difficulty level, tag & Note(if any).

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Please also note any relevant details for your test configuration.

- [x] Test A
- [x] Test B

## Make sure all below guidelines are followed else PR will get Reject:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code so that it is easy to understand
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: Java/PascalsTriangle118.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        if(numRows ==0) return triangle;
+
+        List<Integer> first_row = new ArrayList<>();
+        first_row.add(1);
+        triangle.add(first_row);
+
+        for(int i=1; i<numRows; i++){
+            List<Integer> prev_row = triangle.get(i-1);
+            List<Integer> row = new ArrayList<>();
+            row.add(1);
```
Comment (GouravRusiya30) [inline Java/PascalsTriangle118.java:18]: O(n2) Solution is not acceptable

File: LeetCode-Solutions.iml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+<module type="JAVA_MODULE" version="4">
+  <component name="NewModuleRootManager" inherit-compiler-output="true">
+    <exclude-output />
+    <content url="file://$MODULE_DIR$">
+      <sourceFolder url="file://$MODULE_DIR$/Java" isTestSource="false" />
+    </content>
+    <orderEntry type="inheritedJdk" />
+    <orderEntry type="sourceFolder" forTests="false" />
+  </component>
+</module>
```
Comment (GouravRusiya30) [inline LeetCode-Solutions.iml:11]: Don't add these extra files like .iml, .xml, etc. Only include .java file

File: Java/PascalsTriangle118.java (hunk 1 above)
Comment (akshay399) [reply Java/PascalsTriangle118.java:18]: this is the best time complexity it shows faster than 100% of submissions with 0ms runtime when submitted, and the space complexity is O(N) i will change that now

File: LeetCode-Solutions.iml (hunk 2 above)
Comment (akshay399) [reply LeetCode-Solutions.iml:11]: ok this was added by the ide i will delete them

**Comment by akshay399** [comment]
i did the necessary changes

**Review by GouravRusiya30 — APPROVED** [review]

**Pull request merged**